### ISSUE TITLE
Nagios: Support Windows monitoring

### ISSUE THREAD

**Issue by Willsparker** [adoptium/infrastructure#1718]
ref: #1229 

We don't currently monitor Windows, and have no Nagios stuff for the Windows playbook, so probably best we find a way to add them. AFAIK, it may be doable through Nagios NRPE, however, I need to look into it more..

**Comment by Willsparker** [comment]
Given that it's release week, I don't want to attempt to fix the issues in #1716 , so looking at this.
Broadly speaking, I am mainly finding references to NRPE, as a method of connection to Windows machines from a Nagios Server. We _could_ do it via ssh, but installing an ssh server on Windows machines has been bought up before and rejected. 

I've found a couple guides to monitor Windows machines, so I'll test this on a Nagios Sever running on a local Ubuntu VM to see if I can get it working.

**Comment by aahlenst** [comment]
NRPE and http://www.nsclient.org/ is the way to go for Windows.

**Comment by Willsparker** [comment]
Awesome, thanks :-)

**Comment by Willsparker** [comment]
@sxa has given me https://ci.adoptopenjdk.net/computer/test-azure-win2012r2-x64-1/ to use as a machine to test NRPE / NSClient installation and connecting it to the Nagios Server. I've been able to get a Nagios Server running on a VM, and connected a Windows VM, using NSClient, however that was using `check_nt`, not `check_nrpe`. 

I found the following documents to be quick useful in setting up:
https://assets.nagios.com/downloads/nagiosxi/docs/Enabling-the-NRPE-Listener-in-NSClient-0.4.x-for-Nagios-XI.pdf 
https://support.nagios.com/forum/viewtopic.php?f=7&t=46858&start=10

**Comment by Willsparker** [comment]
Following the two above documents, unfortunately, i am still unable to use `check_nrpe` - with the following error message when attempting to run `check_nrpe -H localhost` 
```
Error: Failed to connect to: localhost:5666 :No connection could be made because the target machine actively refused it.
```
When I `netstat -ab`, I can't find `5666` being listened to, and in the log file:
```
2021-01-25 07:59:00: error:c:\source\master\modules\NRPEServer\NRPEServer.cpp:121: CA not found: C:\Program Files\NSClient++/security/ca.pem (generating a default CA)
2021-01-25 07:59:03: error:B:\master\x64\dist\modules\NRPEServer\module.cpp:37: Exception in Failed to load NRPEServer: : resolve: No such host is known
2021-01-25 07:59:03: error:c:\source\master\service\plugin_manager.cpp:230: Plugin refused to load: NRPEServer
2021-01-25 07:59:05: error:c:\source\master\modules\NSClientServer\NSClientServer.cpp:105: Exception in start: resolve: No such host is known
2021-01-25 07:59:05: error:c:\source\master\service\plugin_manager.cpp:230: Plugin refused to load: NSClientServer
2021-01-25 08:00:50: error:c:\source\master\include\socket/client.hpp:208: Failed to connect to server: No connection could be made because the target machine actively refused it
```
I assume the port isn't listening because the NRPEServer failed to load. 

Found this thread: https://support.nagios.com/forum/viewtopic.php?t=46764 
When followed, the error message becomes, and the port is listening: 
```
2021-01-25 08:47:14: error:c:\source\master\include\socket/connection.hpp:270: Seems we cant agree on SSL: no shared cipher
2021-01-25 08:47:14: error:c:\source\master\include\socket/connection.hpp:271: Please review the insecure options as well as ssl options in settings.
```
Once changing the ` nsclient.ini` file, so insecure `NRPE` connections aren't allowed (i.e. the `insecure` option, in the `/settings/NRPE/server` section), I was able to run the following on the `azure` machine:

```
C:\Program Files\NSClient++>check_nrpe.exe -H 127.0.0.1
I <0.5.2.35 2018-01-28> seem to be doing fine...
```
However, running `check_nrpe` on the Nagios Server gives the following problem:
```
root@nagios /usr/local/nagios/libexec # ./check_nrpe -H 13.68.134.204 -t 30
CHECK_NRPE STATE CRITICAL: Socket timeout after 30 seconds.
```
I double checked and the Nagios Server's IP address is definitely in the `allowed hosts` section of the `nsclient.ini` file. I'll keep digging.

(For safety, I'm going to stop the NSClient Service after I've finished with it, just because I'm unsure of the current security impacts. I'd rather not have an unnecessary open port if I can avoid it)

**Comment by Willsparker** [comment]
I've started the service on `test-azure-win2012r2-x64-1`, and I'm still able to run `check_nrpe.exe -H 127.0.0.1` successfully. This was after having to restart the service once or twice- for some reason the first time I started the service, the command would hang, and the following would be print in the log file:
```
error:c:\source\master\include\socket/server.hpp:258: Failed to handle incoming connection: remote_endpoint: The file handle supplied is not valid
```
Googling says that this may be related to trying to open ports that are already open. Either way, I restarted the service again, checked with `netstat -ab | grep "5666"`, and all seemed well. Not sure what caused it.

As for connecting to that machine with the Nagios Server, still no luck. I was wondering if it may be to do with the Nagios Server's NRPE configuration, however, it is able to run `check_nrpe` with some of the old macos machines we still have (`test-macincloud-macos1010-x64-1`), as some of their services use `nrpe`. 
```
root@nagios /usr/local/nagios/libexec # ./check_nrpe -H 74.80.250.151
NRPE v3.0.1
```
From what I can gather, the Nagios Server has an old version of NRPE installed, and not NSClient++, which may be causing issues:
```
root@nagios /usr/local/nagios/libexec # ./check_nrpe -V

NRPE Plugin for Nagios
Copyright (c) 1999-2008 Ethan Galstad (nagios@nagios.org)
Version: 3.0.1
Last Modified: 09-08-2016
License: GPL v2 with exemptions (-l for more info)
SSL/TLS Available: OpenSSL 0.9.6 or higher required
```
